include/s3select_oper.h

```diff
@@ -1706,6 +1708,10 @@ class base_function
   {//release function-body implementation 
     this->~base_function();
   }
+
+  void extract_timestamp_param_validation(bs_stmt_vec_t*& args);
```

Comment:
/extract_timestamp_param_validation/ and the other two, are not related to base_function class.
base_function is a generic class for ANY operators (aggregate and non-aggregate for one example)

placing these function in /base_function/, means that operators like /sum/ will gain access to these functions while they should not.

a more appropriate 
fn_diff_minute_timestamp -(inherit from)-> base_datediff -(inherit from)-> base_function

the base_datediff will contain the specific functionality related to datediff